helix-view/src/editor.rs

```diff
@@ -315,11 +317,19 @@ impl Editor {
         self._refresh();
     }
 
-    fn new_file_from_document(&mut self, action: Action, mut document: Document) -> DocumentId {
+    /// Generate an id for a new document and register it.
+    fn new_document(&mut self, mut doc: Document) -> DocumentId {
         let id = DocumentId(self.next_document_id);
-        self.next_document_id += 1;
-        document.id = id;
-        self.documents.insert(id, document);
+        // Safety: adding 1 from 1 is fine, probably impossible to reach usize max
+        self.next_document_id =
+            unsafe { NonZeroUsize::new_unchecked(self.next_document_id.get() + 1) };
```

Comment:
While it "probably" won't happen, it technically could, so I'm not sure we should leave it such that it *technically could* cause UB.

Reply:
Technically is possible but if the user have that many documents opened, he probably hit an OOM first rather than this. And chances are the files won't hit that, my filesystem only has like 8881511 and 18446744073709551615 is the max, still a long way to go.

Reply:
We can simplify once `nonzero_ops` gets simplified, by using `saturating_add`.

Reply:
Yeah